src/internal/obj/amazon_client.go

```diff
@@ -404,99 +401,45 @@ func (c *amazonClient) Exists(ctx context.Context, name string) bool {
 	return err == nil
 }
 
-func (c *amazonClient) IsRetryable(err error) (retVal bool) {
-	if strings.Contains(err.Error(), "unexpected EOF") {
-		return true
-	}
-	if strings.Contains(err.Error(), "SlowDown:") {
-		return true
-	}
-
-	var awsErr awserr.Error
-	if !errors.As(err, &awsErr) {
-		return false
-	}
-	for _, c := range []string{
-		storagegateway.ErrorCodeServiceUnavailable,
-		storagegateway.ErrorCodeInternalError,
-		storagegateway.ErrorCodeGatewayInternalError,
-	} {
-		if c == awsErr.Code() {
-			return true
-		}
-	}
-	return false
-}
-
-func (c *amazonClient) IsIgnorable(err error) bool {
-	return false
-}
-
-func (c *amazonClient) IsNotExist(err error) bool {
+func (c *amazonClient) transformError(err error, objectPath string) error {
```

Comment:
We should probably take a pass over these error checks to see if they still make sense and if we should add any new ones. I would like to get rid of the string comparison ones if we can. We can do this later, but it might be good to have a TODO for it.